Data/ByteString.hs

```diff
@@ -47,6 +47,24 @@ module Data.ByteString (
         ByteString,
         StrictByteString,
 
+        -- ** Heap fragmentation
+        -- | Unlike an easy-to-assume expectation, the memory representation of
+        -- 'ByteString' is /distinctly different/ from many other types (such as
+        -- the @Text@ type from the @text@ package). That's necessary to provide
+        -- the guarantees on performance and FFI concerns important to C code.
+        --
+        -- Namely, with GHC the 'ByteString' representation uses /pinned memory/,
+        -- meaning it cannot be moved by GC. This may lead to issues with
+        -- heap fragmentation and wasted space, if the program retains many
+        -- small 'ByteString's live in memory over long durations.
+        --
+        -- While 'ByteString' is indispensable when working with large blobs of
+        -- data and especially when interfacing with native C libraries — be
+        -- sure to also check the 'Data.ByteString.Short.ShortByteString' type.
+        -- As a type backed by /unpinned/ memory, @ShortByteString@ is more
+        -- similar to @Text@ on the heap, completely avoids fragmentation issues,
```

Comment:
If you take my suggestion to remove the reference to `text` from the first paragraph, this reference to `Text` should probably become a parenthetical: "`ShortByteString` (like `Data.Text.Text`) completely avoids..."

Reply:
I see a thumbs-up and a resolved conversation, but no change from the previous version on this point. Did you forget to commit this change, or did you change your mind?

Reply:
The latter — I figured that for a (prospective) user of bytestring package, the Text type from `text` package is highly likely not news; yet if it is, the potential for confusion is low, as there aren't many types with unqualified name `Text` in Haskell at present. There're aliases of course, but in 3 web-search engines I checked the top result for "haskell Text" leads to the `text` package.

So, left it as-is, to avoid awkward wrapping, and to not overload the sentence with 2nd-order tangential detail.

The exact reference to `text` package is still discoverable, by following the link to `ShortByteString`:

https://github.com/haskell/bytestring/blob/39f40116a4adf8a3296067d64bd00e1a1e5e15bd/Data/ByteString/Short.hs#L64-L66

Reply:
The reason I wanted the parenthetical is mostly that it makes more clear that `Text` uses unpinned memory and that is the way they are similar.

I agree that nearly all readers will be familiar with or easily find the `text:Data.Text.Text` type without any disambiguation.

(Awkward wrapping is only visible to readers of our source code. It doesn't affect the rendered documentation.)

Reply:
Parenthetical added, `@ShortByteString@ behaves similarly to @Text@ (from the @text@ package) on the heap` — pretty clear IMO, seems pretty hard to mis-read